activation/post.go

```diff
@@ -360,14 +366,16 @@ func (mgr *PostSetupManager) StopSession(deleteFiles bool) error {
 }
 
 // GenerateProof generates a new Post.
-func (mgr *PostSetupManager) GenerateProof(challenge []byte) (*types.Post, *types.PostMetadata, error) {
+func (mgr *PostSetupManager) GenerateProof(challenge []byte, commitmentAtx types.ATXID) (*types.Post, *types.PostMetadata, error) {
 	state := mgr.getState()
 
 	if state != postSetupStateComplete {
 		return nil, nil, errNotComplete
 	}
 
-	prover, err := proving.NewProver(config.Config(mgr.cfg), mgr.LastOpts().DataDir, mgr.id)
+	// TODO(mafa): id field in post package should be renamed to commitment otherwise error messages are confusing
+	commitment := sha256.Sum256(append(mgr.id, commitmentAtx.Bytes()...))
```

Comment:
have a method for calculation commitment bytes? as is, if we were to change hashing algo, we need to change it in many places...

Reply:
Yes, I will add a method.

Imho this should be part of the PoST library, just like the parameter taking the commitment should be called commitment and not `id` (leads to confusing error messages in the worst case).

Reply:
I changed it by using the `hash` package.